apps/api/src/app/notification-groups/usecases/update-notification-group/update-notification-group.usecase.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { NotificationGroupRepository, NotificationGroupEntity } from '@novu/dal';
+import { UpdateNotificationGroupCommand } from './update-notification-group.command';
+import { NotificationGroupResponseDto } from '../../dtos/notification-group-response.dto';
+
+@Injectable()
+export class UpdateNotificationGroup {
+  constructor(private notificationGroupRepository: NotificationGroupRepository) {}
+
+  async execute(command: UpdateNotificationGroupCommand) {
+    const { id, environmentId, name } = command;
+
+    const item = await this.notificationGroupRepository.findOne({
+      _environmentId: environmentId,
+      _id: id,
+    });
+
+    if (!item) {
+      throw new NotFoundException();
+    }
```

Comment:
Instead of doing this you can compose this use case by using the `GetNotificationGroup` use case you created as there you are already handling the situation of a group not found. Therefore if the logic would need to be changed to manage that use case you would only have to change it in one place as this use case depends on the safe guard of a group existing.